### PR TITLE
LPS-153790 Poshi Automation for Bug | master

### PR DESCRIPTION
**Pull request by ozysouza**
https://issues.liferay.com/browse/LPS-153790

@david-gutierrez-mesa 
@Tim-Cao

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/MockMock.path [hunk 1]
```diff
@@ -10,6 +10,16 @@
...
 <tbody>
+<tr>
+	<td>MOCKMOCK_CONTENT_ADDED_DIFF</td>
+	<td>//div[@class='well']//span[contains(@id,'added-diff')]</td>
+	<td></td>
+</tr>
+<tr>
+	<td>MOCKMOCK_CONTENT_REMOVED_DIFF</td>
+	<td>//div[@class='well']//span[contains(@style,'line-through') and contains(@id,'removed-diff')]</td>
+	<td></td>
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/paths/pathlib/uisections/MockMock.path:21]: Merge into one
```
	<td>MOCKMOCK_HTML_BODY_FORMATTED_FIELD_DIFF</td>
	<td>//div[@name='bodyHTML_Formatted']//div[@class='well']//span[contains(@changeid,'${changeType}-diff')]</td>
```

File: portal-web/test/functional/com/liferay/portalweb/macros/MockMock.macro [hunk 2]
```diff
@@ -235,6 +245,24 @@ definition {
 		MockMock._gotoMail();
 	}
 
+	macro viewEmailDiff {
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/MockMock.macro:238]: viewFormattedHTMLTextDIFF

File: portal-web/test/functional/com/liferay/portalweb/macros/MockMock.macro [hunk 3]
```diff
@@ -48,6 +48,16 @@ definition {
...
 
+	macro _viewEmailDiff {
+		AssertTextEquals(
+			locator1 = "MockMock#MOCKMOCK_CONTENT_ADDED_DIFF",
+			value1 = "${newContent}");
+
+		AssertTextEquals(
+			locator1 = "MockMock#MOCKMOCK_CONTENT_REMOVED_DIFF",
+			value1 = "${oldContent}");
+	}
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/MockMock.macro:58]: No need

File: portal-web/test/functional/com/liferay/portalweb/macros/MockMock.macro [hunk 4]
```diff
@@ -235,6 +245,24 @@ definition {
...
+
+		MockMock._viewMailFromInbox(
+			asset = "${asset}",
+			recipient = "${recipient}",
+			sender = "${sender}");
+
+		AssertClick.assertPartialTextClickAt(
+			key_columnEntry = "${asset}",
+			locator1 = "MockMock#MOCKMOCK_SUBJECT_FIELD_LINK",
+			value1 = "${asset}");
```
Comment (Tim-Cao) [inline portal-web/test/functional/com/liferay/portalweb/macros/MockMock.macro:249]: MockMock._gotoMailMessage(asset = "${asset}");

**Comment by Tim-Cao** [comment]
Hi @ozysouza, would you mind updating according to comments? Thanks!

File: portal-web/test/functional/com/liferay/portalweb/macros/journal/WebContent.macro [hunk 5]
```diff
@@ -1211,6 +1211,45 @@ definition {
...
+		if (isSet(disableCheckbox)) {
+			Uncheck(
+				checkboxName = "${disableCheckbox}",
+				locator1 = "Checkbox#ANY_CHECKBOX");
+		}
+
+		if (isSet(subjectTitle)) {
+			Type(
+				key_fieldLabel = "${subjectId}",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
```
[the remark this answers is not in the document]
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/macros/journal/WebContent.macro:1228]: Same comment than above, the path finds 14 results. Is it okay?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentWithUserNotifications.testcase [hunk 6]
```diff
@@ -129,6 +129,98 @@ definition {
...
+				workflowTask = "Review");
+		}
+
+		task ("Then the user could view the article diffs on granted email notification") {
+			MockMock.viewEmailDiff(
+				asset = "Web Content: Article Approval Granted",
+				newContent = "Where does it come from?",
+				oldContent = "Web Content Content",
+				recipient = "Test Test",
+				sender = "Test Test");
```
[the remark this answers is not in the document]
Comment (ozysouza) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentWithUserNotifications.testcase:220]: @Tim-Cao, about the recipient  name, who receives the Granted e-mail is the test test and not the new user. I made I call with Laszlo by the time asking exactly it, if the user should receives the granted e-mail, he told me it was the expected by the time.

**Comment by ozysouza** [comment]
> Hi @ozysouza, would you mind updating according to comments? Thanks!

Hi @Tim-Cao, PR updated! Could you please have a look at? I wrote some comments just to clarify some doubts. 

Many thanks for the heads up to improve the macro/path, it is way better.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentWithUserNotifications.testcase (hunk 6 above)
[the remark this answers is not in the document]
Comment (Tim-Cao) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/wem/journal/WebContentWithUserNotifications.testcase]: You're right!

**Comment by Tim-Cao** [comment]
Small changes @ozysouza https://github.com/liferay-echo/liferay-portal/pull/9511

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**